src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -1082,8 +1082,17 @@ internal async Task UpdateDetailPaneAsync(CancellationToken cancellationToken)
 
                 EmitSearchSelectionTelemetry(selectedItem);
 
-                await _detailModel.SetCurrentPackageAsync(selectedItem, _topPanel.Filter, () => _packageList.SelectedItem);
-                _detailModel.SetCurrentSelectionInfo(selectedIndex, recommendedCount, _recommendPackages, selectedItem.RecommenderVersion);
+                try
```

Comment:
I am working in this spot in my branch, and I almost added a `cancellationToken.ThrowIfCancelled` call here. Should you go ahead and do similar, since your PR focuses on cancellation.

We don't use the token, so maybe we should at least allow it to throw. That's my thought.

Reply:
let's talk through with a quick call. will see if you are available.

Reply:
we talked. we agreed to plumb through this real cancellation token through. see latest commit.